src/main/java/com/spotify/reaper/resources/view/NodesStatus.java

```diff
@@ -0,0 +1,208 @@
+      Optional<String> rack = Optional.absent();
+      Optional<String> status = Optional.absent();
+      Optional<Double> severity = Optional.absent();
+      Optional<String> releaseVersion = Optional.absent();
+      Optional<String> tokens = Optional.absent();
+      Optional<Double> load = Optional.absent();
+      
+      String endpointString = strEndpoints[i];
+      
+      matcher = endpointNamePattern.matcher(endpointString);
+      if(matcher.find()) {
+        endpoint = Optional.fromNullable(matcher.group(1));
+      }
+      
+      matcher = endpointStatusPattern.matcher(endpointString);
+      if(matcher.find()) {
+        status = Optional.fromNullable(matcher.group(3));
+      }
+      
+      matcher = endpointDcPattern.matcher(endpointString);
+      if(matcher.find()) {
+        dc = Optional.fromNullable(matcher.group(3));
+      }
+      
+      matcher = endpointRackPattern.matcher(endpointString);
+      if(matcher.find()) {
+        rack = Optional.fromNullable(matcher.group(3));
+      }
+      
+      matcher = endpointSeverityPattern.matcher(endpointString);
+      if(matcher.find()) {
+        severity = Optional.fromNullable(Double.parseDouble(matcher.group(3)));
+      }
+      
+      matcher = endpointLoadPattern.matcher(endpointString);
+      if(matcher.find()) {
+        load = Optional.fromNullable(new BigDecimal(matcher.group(3)).doubleValue());
+        totalLoad+=load.or(0.0);
+      }
+      
+      matcher = endpointReleasePattern.matcher(endpointString);
+      if(matcher.find()) {
+        releaseVersion = Optional.fromNullable(matcher.group(3));
+      }
+      
+      matcher = endpointHostIdPattern.matcher(endpointString);
+      if(matcher.find()) {
+        hostId = Optional.fromNullable(matcher.group(3));
+      }
+      
+      matcher = endpointTokensPattern.matcher(endpointString);
+      if(matcher.find()) {
+        tokens = Optional.fromNullable(matcher.group(2));
+      }
+      
+      
+      
+      EndpointState endpointState = new EndpointState(endpoint.or("Not available"), hostId.or("Not available"), dc.or("Not available"), rack.or("Not available"), status.or("Not available"), severity.or(0.0),
+          releaseVersion.or("Not available"), tokens.or("Not available"), load.or(0.0));
+      
+      endpoints.add(endpointState);
+    }
+    
+    Map<String, Map<String, List<EndpointState>>> endpointsByDcAndRack = Maps.newHashMap();
+    Map<String, List<EndpointState>> endpointsByDc = endpoints.stream().collect(Collectors.groupingBy(EndpointState::getDc, Collectors.toList()));
+    
+    for(String dc:endpointsByDc.keySet()) {
+      Map<String, List<EndpointState>> endpointsByRack = endpointsByDc.get(dc).stream().collect(Collectors.groupingBy(EndpointState::getRack, Collectors.toList()));
+      endpointsByDcAndRack.put(dc, endpointsByRack);
+    }
+    
+    GossipInfo gossipInfo = new GossipInfo(sourceNode ,endpointsByDcAndRack, totalLoad);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "gossipInfo". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)